src/pages/therapists/index.tsx

```diff
@@ -1,34 +1,140 @@
+    setCurrentPage(page);
+  };
+
+  const renderLoadingState = () => (
+    <Grid container spacing={2}>
+      {[...Array(8)].map(() => (
+        <Grid item xs={12} sm={6} md={4}>
+          <br />
+          <br />
+          <Box
+            sx={SkeletonBoxStyle}
+          >
+            <Skeleton variant="rectangular" width="100%" height={180} animation="wave" />
+            <Box p={2}>
+              <Skeleton variant="text" width="50%" height={20} animation="wave" />
+              <Skeleton variant="text" width="70%" height={16} animation="wave" sx={{ mt: 1 }} />
+            </Box>
+          </Box>
+        </Grid>
+      ))}
+    </Grid>
+  );
+
+  const renderTherapists = () => (
+    <Grid container spacing={2}>
+      {therapists.map((therapist:TherapistCardType) => (
+        <GridCard therapist={therapist} key={therapist.userId} />
+      ))}
+    </Grid>
+  );
+
+  const renderNoTherapistsMessage = () => (
+    <Alert severity="info" sx={{ mt: 4 }}>
+      No therapists found.
+    </Alert>
+  );
+
+  return (
+    <>
+      <Container>Therapist Page</Container>
+      <Box sx={{ flexGrow: 1 }}>
+        <Container>
+          <Grid
+            item
+            xs={12}
+            md={8}
+            sx={{
+              display: 'flex',
+              justifyContent: 'center',
+              alignItems: 'center',
+              mt: 4,
+            }}
+          >
+            <InputBase
+              placeholder="Search..."
+              inputProps={{ 'aria-label': 'search' }}
+              value={searchQuery}
+              onChange={handleSearch}
+              style={SearchBoxStyle}
+              sx={{
+                '& .MuiInputBase-input': {
+                  fontSize: '1rem',
+                  color: 'inherit',
+                  padding: '10px 12px',
+                },
+                '& .MuiInputBase-input::placeholder': {
+                  color: '#999',
+                  opacity: 1,
+                },
+              }}
+            />
+            <IconButton aria-label="search" sx={{ ml: 1 }}>
+              <SearchIcon />
+            </IconButton>
+          </Grid>
+        </Container>
+      </Box>
       <Container>
-        <Grid item xs={6} md={8} style={{ marginTop: '20px' }}>
-          <InputBase
-            placeholder="Search..."
-            inputProps={{ 'aria-label': 'search' }}
+        {loading && renderLoadingState()}
+        {!loading && therapists.length === 0 && renderNoTherapistsMessage()}
+        {!loading && therapists.length > 0 && renderTherapists()}
+        <Box sx={{ display: 'flex', justifyContent: 'center', marginTop: '20px' }}>
+          <Pagination
+            count={totalPages}
+            page={currentPage}
+            onChange={handlePageChange}
+            color="primary"
           />
-          <IconButton aria-label="search">
-            <SearchIcon />
-          </IconButton>
-        </Grid>
+        </Box>
       </Container>
```

Comment:
What about moving this container to a separate component and passing loading and therapists as a props?